src/external/dftk_to_wannier90.jl

```diff
@@ -0,0 +1,605 @@
+        write(f,"end kpoints"*"\n"^2)
+
+        @info "File generated.",f
+    end
+    
+    
+end
+
+
+"""
+Read the .nnkp file provided by the preprocessing routine of Wannier90 (i.e. "wannier90.x -pp prefix")
+Returns: 
+1) the array 'nn_kpts' of k points, they respective neirest neighbours and associated shifing vectors (non zero if the neighbour of is located in another cell).
+2) the number 'nn_num' of neighbours per k point. 
+2) the array 'projs' of projections. Each line contains the informations for one projection :  [center],[quantum numbers: l,mr,r],[ref. z_axis],[ref. x_axis],α = Z/a
+"""
+function read_nnkp_file(prefix::String)
+
+    ## Check the presence of the file
+    if !isfile("$prefix.nnkp")
+        error("file $prefix.nnkp not found.")   
+    end
+
+    ## Read the file
+    file = open("$prefix.nnkp")
+    @info "Reading nnkp file", file 
+    ln = [string(l) for l in eachline(file)]
+    close(file)
+
+
+    ## Extract nnkp block 
+    i_nn_kpts = findall(x-> endswith(x,"nnkpts"),ln)                               #Indices of the first and last line of the nnkpts block
+    @assert size(i_nn_kpts,1) == 2
+    char_to_vec(line) = [parse.(Int,x) for x in split(line,' ',keepempty=false)]
+    
+    nn_num = parse(Int64,ln[i_nn_kpts[1]+1])                                       #number of neighbours per k points
+    nn_kpts =  [ char_to_vec(l) for l in ln[i_nn_kpts[1]+2:i_nn_kpts[2]-1] ]       
+
+
+
+    ## Extract projections block : NEEDED FOR WIN GUESSES
+    i_projs = findall(x-> endswith(x,"projections"),ln)                                  #Indices of the first and last line of the projection block
+    @assert size(i_projs,1) == 2
+    raw_projs = [split(ln[i],' ',keepempty = false) for i in i_projs[1]+1:i_projs[2]-1]  #data in strings
+
+    # reshape so that one line gives all infos about one projection
+    n_projs = parse(Int,only(popfirst!(raw_projs)))
+    @assert(n_projs == size(raw_projs,1)/2)
+    raw_projs = reshape(raw_projs,(2,n_projs))
+
+    # Parse in the format proj = [ [[center],[l,mr,r],[z_axis],[x_axis], α], ... ]
+    projs = []
+    for j in 1:n_projs
+        center = [parse(Float64,x) for x in raw_projs[1,j][1:3]]
+        quantum_numbers = [parse(Int,x) for x in raw_projs[1,j][4:end] ]
+        z_axis = [parse(Float64,x) for x in raw_projs[2,j][1:3]]
+        x_axis = [parse(Float64,x) for x in raw_projs[2,j][4:6]]
+        α = parse(Float64,raw_projs[2,j][7])
+        push!(projs, [center,quantum_numbers,z_axis,x_axis,α])
+    end
+    
+    nn_kpts,nn_num,projs    
+end
+
+
+
+"""
+Take scf_res, the result of a self_consistent_field calculation and writes eigenvalues (in eV) in a format readable by Wannier90.
+"""
+function generate_eig_file(prefix::String,scf_res)
+    Ha_to_eV = 1/(unit_to_au(:eV))
+
+    eigvalues = scf_res.eigenvalues
+    k_size = size(eigvalues,1)
+    n_bands = size(eigvalues[1],1)
+
+    #write file
+    open("$prefix.eig","w") do f
+        for k in 1:k_size
```

Comment:
Try to use the `enumerate` iterator for such things. It gives you an index and the object at the same time, which is typically much nicer to read.